### PR TITLE
os/bluestore: fixing some non-critical bugs for HybridAllocator

### PR DESCRIPTION
**Pull request by ifed01**
Signed-off-by: Igor Fedotov <igor.fedotov@croit.io>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ifed01** [comment]
@aclamk - ping ;)

**Comment by ifed01** [comment]
unstale

**Comment by rzarzynski** [comment]
@aclamk, @ifed01: is this PR still relevant?

**Comment by ifed01** [comment]
> @aclamk, @ifed01: is this PR still relevant?

yep!

File: src/os/bluestore/HybridAllocator.cc [hunk 1]
```diff
@@ -39,68 +39,52 @@ int64_t HybridAllocator::allocate(
...
-      if (orig_size) {
-        local_extents.insert(
-          local_extents.end(), ++orig_pos, extents->end());
-        extents->resize(orig_size);
-      } else {
-        extents->swap(local_extents);
-      }
-      _release(local_extents);
+      // allocator shouldn't return new extents on error
+      ceph_assert(orig_size == extents->size());
```
Comment (aclamk) [inline src/os/bluestore/HybridAllocator.cc:77]: Here you assume that allocator will not return partial request.
This seems true for current implementations.
However we should write a contract for Allocator interface that states that. And UT that checks it.

**Review by aclamk — APPROVED** [review]

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomcPuCOnhYL1841-wip-yuri5-testing-2023-10-02-1105-old-wip-yuri5-testing-2023-09-27-0959

**Comment by yuriw** [comment]
jenkins test this please

**Comment by ifed01** [comment]
jenkins test make check

**Pull request closed without merging**